src/ecs/saveload/ser.rs

```diff
@@ -0,0 +1,135 @@
+use std::error::Error;
+use std::marker::PhantomData;
+
+use serde::ser::{self, Serialize, Serializer, SerializeSeq};
+
+use shred::{Resources, ResourceId, SystemData};
+use specs::{Entities, FetchMut, Join, ReadStorage, WriteStorage};
+
+use super::{Components, EntityData, Storages};
+use super::marker::{Marker, MarkerAllocator};
+
+
+/// Serialize entities
```

Comment:
Since this a public function, could we get a more in-depth description on what it does?